R/print.data.table.R

```diff
@@ -164,3 +186,39 @@ shouldPrint = function(x) {
 #   as opposed to printing a blank line, for excluding col.names per PR #1483
 cut_top = function(x) cat(capture.output(x)[-1L], sep = '\n')
 
+# to calculate widths of data.table for PR #4074
+# gets the width of the data.table at each column
+#   and compares it to the console width
+dt_width = function(x, class, row.names, col.names) {
+  widths = apply(nchar(x, type='width'), 2L, max)
+  if (class) widths = ifelse(widths < 6L, 6L, widths)
```

Comment:
I really hate `base::ifelse` 😅 I would do `widths = pmax(widths, 6L)`

Reply:
Is some of that hate from its unsafe behavior around coercion? Will update.

Reply:
Can see here... it's "slow" though probably not noticeable at the size of `ncol`:

https://github.com/Rdatatable/data.table/issues/2677